### ISSUE TITLE
CheckMesh Maximum Skewness

### ISSUE THREAD

**Issue by MikkiSeidenschnur** [ladybug-tools/butterfly#237]
Hi Guys

I know this is probably more appropriate on the CFDOnline Forum, but I think that @TheodoreGalanos or @thinklikeanarchitect might be able to answer this one.

I'm wondering how the MaxInternalSkewness for checkMesh can be higher than one. For instance in [https://cfd.direct/openfoam/user-guide/snappyHexMesh/#x26-1930005.4.8](url) they say that the maximum allowed Skewness is 4. 

What I'm asking is: How does checkMesh for SHM actually calculate the internalCellSkewness? Because I cannot for the life of me find out why it is possible to have something higher than 1 :). 

I hope you can help me out here, let me know if I'm not formulating the question to an understandable level. 

Best regards
Mikki Seidenschnur

**Comment by TheodoreGalanos** [comment]
Hi @MikkiSeidenschnur If I understand correctly what you are referring to take a look at page 51.

https://openfoamwiki.net/images/f/f0/Final-AndrewJacksonSlidesOFW7.pdf

**Comment by MikkiSeidenschnur** [comment]
@TheodoreGalanos Yes, so that is actually the part where I'm in doubt - So that's what I'm getting at: How can the skewness actually be higher than 1, because for me that would mean that the magnitude(di)>magnitude(ci)? Is that even possible?

**Comment by hrobeers** [comment]
The skewness value reported by checkMesh is calculated differently than is done by most others (paraview, fluent, ...).

For the same structured mesh I got the following max skewness:
- checkMesh: 0.90
- paraview: 0.49
